saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/metadata/OpenSamlMetadataResolver.java

```diff
@@ -111,6 +113,9 @@ private EntityDescriptor entityDescriptor(RelyingPartyRegistration registration)
 		SPSSODescriptor spSsoDescriptor = buildSpSsoDescriptor(registration);
 		entityDescriptor.getRoleDescriptors(SPSSODescriptor.DEFAULT_ELEMENT_NAME).add(spSsoDescriptor);
 		this.entityDescriptorCustomizer.accept(new EntityDescriptorParameters(entityDescriptor, registration));
+		if (this.signMetadata) {
```

Comment:
Instead of signing each entity descriptor, please sign the entire payload. I believe this will mean that you'll call `sign` separately [here](https://github.com/spring-projects/spring-security/pull/14916/files#diff-81a8047cc4069e0ae01dcb90c3e43e2c829bd24006e7f3df32f72310917b4eb8L101) and [here](https://github.com/spring-projects/spring-security/pull/14916/files#diff-81a8047cc4069e0ae01dcb90c3e43e2c829bd24006e7f3df32f72310917b4eb8L105).

Reply:
Hi @jzheaux ! Thanks for your feedback. In the case of signing `EntitiesDescriptor` with a list of children, it is not clear which SP and IP pair to use. In other words, which of the `relyingPartyRegistrations` should be passed to the `sign` method?

Reply:
Good catch! Let's leave it as it is.